src/libsyntax_ext/deriving/cmp/partial_eq.rs

```diff
@@ -52,7 +55,10 @@ pub fn expand_deriving_partial_eq(cx: &mut ExtCtxt,
                     _ => cx.span_bug(span, "not exactly 2 arguments in `derive(PartialEq)`")
                 };
 
-                let eq = cx.expr_binary(span, BinOpKind::Ne, self_f, other_f.clone());
+                let eq = cx.expr_binary(span,
+                                        BinOpKind::Ne,
+                                        cx.expr_addr_of(span, self_f),
+                                        cx.expr_addr_of(span, other_f.clone()));
```

Comment:
In the past we've had subtle bugs where fields didn't actually implement some trait you were deriving but it worked anyway, so I'm curious why the extra `&` is needed here? In theory that should all happen automatically, right?